lib/view_component/base.rb

```diff
@@ -223,6 +223,9 @@ def content_evaluated?
       @_content_evaluated
     end
 
+    # Configure if ActiveSupport notifications are enabled for all ViewComponents.
+    mattr_accessor :instrumentation_enabled, instance_writer: false, default: false
```

Comment:
Should this be included as part of the `Instrumentation` module instead of being defined directly on base?

Since we're utilizing Rails config I think we can use something like `Rails.application.config.view_component.instrumentation_enabled` when needed?